File: src/main/java/com/docs/viewer/global/common/dto/response/ResponseHandler.java
package com.docs.viewer.global.common.dto.response;

import com.docs.viewer.global.error.dto.response.ErrorResponse;
import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class ResponseHandler {

    private int status;

    private String message;

    public ResponseHandler(int status, String message) {
        this.status = status;
        this.message = message;
    }

    public static ResponseHandler error(ErrorResponse response) {
        return ResponseHandler.builder()
                .status(response.getStatus())
                .message(response.getMessage())
                .build();
    }
}

File: dotnet-docs-viewer/Common/Dto/Response/ResponseHandler.cs
using Newtonsoft.Json;
using WebApp.Common.Error;

namespace WebApp.Common.Dto
{
    [JsonObject]
    public abstract class ResponseHandlerBase
    {
        public virtual int Status { get; protected set; }
        public virtual string Message { get; protected set; } = null!;
    }

    public class ResponseHandler : ResponseHandlerBase
    {
        public static ResponseHandler Success(int status, string message)
        {
            return new ResponseHandler
            {
                Status = status,
                Message = message
            };
        }
        
        public static ResponseHandler Error(ErrorResponse response)
        {
            return new ResponseHandler
            {
                Status = response.Status,
                Message = response.Message
            };
        }
    }
}